client/verta/verta/dataset/_s3.py

```diff
@@ -93,12 +93,23 @@ def _get_s3_loc_metadata(cls, s3_loc):
             six.raise_from(e, None)
         s3 = boto3.client('s3')
 
-        # TODO: handle prefixes
-        if s3_loc.key is None:
-            # TODO: handle `bucket_name` not found
-            for obj in s3.list_object_versions(Bucket=s3_loc.bucket)['Versions']:
-                if obj['IsLatest']:
-                    yield cls._get_s3_obj_metadata(obj, s3_loc.bucket, obj['Key'])
+        if (s3_loc.key is None  # bucket
+                or s3_loc.key.endswith('/')):  # folder
```

Comment:
What happens if I send a folder that doesn't end with `/`? For example, `s3://my-bucket/my-folder`. I image the `prefix` argument work work just the same. In fact, the prefix argument might work the same even if that's a file.

Reply:
Handling prefixes more broadly is already a separate TODO, and I didn't allocate bandwidth this sprint to do that. This PR was just meant to fix a bug that was causing issues downstream.

It's a _substantially_ more complicated change. If a user has a folder `foo/` and an object `foobar.csv` at the same level, then sending `"foo"` as a prefix is going to pick up that object, which is wrong. I have some ideas for how to do it, but it's not what I planned to do here.

Reply:
Makes sense. Seems messier indeed.

Reply:
What would happen if I provide `s3://my-bucket/my-folder` today? Will we still get NaN in the values?

Reply:
Good question!
Both before _and after_ this fix, the Client raises a  
```
ClientError: An error occurred (404) when calling the HeadObject operation: Not Found
```
from boto3